packages/wrapped-keys/src/lib/wrapped-keys.ts

```diff
@@ -0,0 +1,216 @@
+import { CHAIN_ETHEREUM, ENCRYPTED_PRIVATE_KEY_ENDPOINT } from './constants';
+import { decryptToString, encryptString } from '@lit-protocol/encryption';
+import { logError } from '@lit-protocol/misc';
+import {
+  getFirstSessionSig,
+  getPkpAccessControlCondition,
+  getPkpAddressFromSessionSig,
+} from './utils';
+import {
+  LitMessage,
+  LitTransaction,
+  ExportPrivateKeyParams,
+  ExportPrivateKeyResponse,
+  ImportPrivateKeyParams,
+  ImportPrivateKeyResponse,
+  SignWithEncryptedKeyParams,
+} from './interfaces';
+
+export async function importPrivateKey({
+  pkpSessionSigs,
+  privateKey,
+  litNodeClient,
+}: ImportPrivateKeyParams): Promise<string> {
+  const firstSessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(firstSessionSig);
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  const { ciphertext, dataToEncryptHash } = await encryptString(
+    {
+      accessControlConditions: allowPkpAddressToDecrypt,
+      dataToEncrypt: privateKey,
+    },
+    litNodeClient
+  );
+
+  const data = {
+    ciphertext,
+    dataToEncryptHash,
+  };
+
+  try {
+    const response = await fetch(ENCRYPTED_PRIVATE_KEY_ENDPOINT, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        pkpsessionsig: JSON.stringify(firstSessionSig),
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
```

Comment:
we should check the `Content-Type` and before we assume `text` as the serializer. This might not actually matter with this particular api

Reply:
@MaximusHaximus I think our Lambda will always return a test response?